src/executables/util_service.c

```diff
@@ -2443,19 +2443,23 @@ process_javasp_start (const char *db_name, bool process_window_service)
 	      status = NO_ERROR;
 	      break;
 	    }
-	  else if (javasp_status == JAVASP_SERVER_STOPPED)
+	  else
 	    {
 	      sleep (1);	/* wait to start */
 	      waited_secs++;
 
-	      util_log_write_errstr ("Waiting for javasp server to start... (%d / %d)\n", waited_secs, wait_timeout);
-	    }
-	  else
-	    {
-	      if (waited_secs > 3)
+	      if (javasp_status == JAVASP_SERVER_STOPPED)
 		{
-		  /* invalid database name or failed to open info file, wait upto 3 seconds */
-		  break;
+		  util_log_write_errstr ("Waiting for javasp server to start... (%d / %d)\n", waited_secs,
+					 wait_timeout);
+		}
+	      else
+		{
+		  if (waited_secs > 3)
```

Comment:
is it correct that if wait_timeout is 30, waited_secs is 3 not 30 ?

Reply:
When starting the JVM, is_javasp_running () can return STOPPED and ERROR.
This is to avoid waiting too long for the user to repeatedly encounter errors when running the JVM.
In the case of ERROR, the reason for setting 3 seconds without exiting immediately is that when the JVM is executed for the first time, Sometimes, ERROR is returned at first for some abnormal reason and then changed to the STOPPED status. And the JVM succeeds to be executed after a certain period of time. This is a kind of heuristic approach.

Reply:
OK, i got it.